### PR TITLE
Make dot output simpler by default

### LINKED ISSUE #339: simplified dot output
**Is your feature request related to a problem? Please describe.**

In order to give a useful graph for display in the current UI, we need a simplified dot file 

**Describe the solution you'd like**

`renku log --simple --format=dot`

This would mean removing all `type` predicates and associated objects; all associations need to be collapsed and the direct edge between the two entities created; `location` predicates need to be folded into the object node.

### PR DESCRIPTION
**Pull request by rokroskar**
# Description

Provides an option for a simplified dot graph generation showing only inputs, outputs, and processes.

Fixes #339 

## Type of change

Please select relevant options.

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have run ``./run-tests.sh`` locally.

## Example output

simple

simple

### CODE REVIEW

**Comment by rokroskar** [comment]
@jirikuncar did the refactor we discussed, I think it's gtg

File: renku/cli/log.py [hunk 1]
```diff
@@ -86,9 +86,15 @@
     default=False,
     help='Display commands without output files.'
 )
+@click.option(
```
Comment (jirikuncar) [inline renku/cli/log.py]: I would rather see options:

```
dot
dot-landscape
dot-full
dot-full-landscape
```

Comment (jirikuncar) [reply renku/cli/log.py]: Keeping in mind that `dot` should be reasonable for UI.

File: renku/cli/log.py [hunk 2]
```diff
@@ -86,9 +86,15 @@
...
+    '--simple',
+    is_flag=True,
+    default=False,
+    help='Generate a simplified graph.'
+)
 @click.argument('paths', type=click.Path(exists=True), nargs=-1)
 @pass_local_client
-def log(client, revision, format, no_output, paths):
+def log(client, revision, format, no_output, simple, paths):
```
Comment (jirikuncar) [inline renku/cli/log.py]:
```suggestion
def log(client, revision, format, no_output, paths):
```

File: renku/cli/log.py [hunk 3]
```diff
@@ -101,4 +107,4 @@ def log(client, revision, format, no_output, paths):
     # NOTE shall we warn when "not no_output and not paths"?
     graph.build(paths=paths, revision=revision, can_be_cwl=no_output)
 
-    FORMATS[format](graph)
+    FORMATS[format](graph, simple=simple)
```
Comment (jirikuncar) [inline renku/cli/log.py]:
```suggestion
    FORMATS[format](graph)
```

File: renku/cli/_format.py [hunk 4]
```diff
@@ -41,7 +43,7 @@ def _jsonld(graph, format, *args, **kwargs):
     return json.dumps(output, indent=2)
 
 
-def dot(graph):
+def dot(graph, simple=False, landscape=True):
```
Comment (jirikuncar) [inline renku/cli/_format.py]:
```suggestion
def dot(graph, simple=True, landscape=True):
```

File: renku/cli/_format.py [hunk 5]
```diff
@@ -55,20 +57,130 @@ def dot(graph):
...
     )
+
+    g.bind('prov', 'http://www.w3.org/ns/prov#')
+    g.bind('wfdesc', 'http://purl.org/wf4ever/wfdesc#')
+    g.bind('wf', 'http://www.w3.org/2005/01/wf/flow#')
+    g.bind('wfprov', 'http://purl.org/wf4ever/wfprov#')
+
+    if simple:
+        _rdf2dot_simple(g, sys.stdout, landscape)
+        return
```
Comment (jirikuncar) [inline renku/cli/_format.py]:
```suggestion
    else:
```

File: renku/cli/_format.py [hunk 6]
```diff
@@ -55,20 +57,130 @@ def dot(graph):
...
     rdf2dot(g, sys.stdout)
```
Comment (jirikuncar) [inline renku/cli/_format.py]:
```suggestion
        rdf2dot(g, sys.stdout)
```

File: renku/cli/_format.py [hunk 7]
```diff
@@ -17,10 +17,12 @@
 # limitations under the License.
 """Serializers for graph data."""
 
+import functools
+
 import click
 
 
-def ascii(graph):
+def ascii(graph, **kwargs):
```
Comment (jirikuncar) [inline renku/cli/_format.py]: I would remove the `**kwargs` since we don't have generic options that would apply to all formatters.

**Comment by rokroskar** [comment]
@jirikuncar please re-review, I think it's gtg.

File: renku/cli/_format.py (hunk 7 above)
Comment (jirikuncar) [reply renku/cli/_format.py]: @rokroskar do we still need `**kwargs` everywhere?

Comment (rokroskar) [reply renku/cli/_format.py]: oops no, I forgot to remove them there.

**Pull request merged**